sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,288 @@ mod test_kubernetes_bootstrap_token {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// EvictionHardKey represents a string that contains a valid Kubernetes eviction hard signal.
+/// Available eviction hard signals: https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct EvictionHardKey {
+    inner: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum EvitionSignal {
+    #[serde(rename = "memory.available")]
+    MemoryAvailable,
+    #[serde(rename = "nodefs.available")]
+    NodefsAvailable,
+    #[serde(rename = "nodefs.inodesFree")]
+    NodefsInodesFree,
+    #[serde(rename = "imagefs.available")]
+    ImagefsAvailable,
+    #[serde(rename = "imagefs.inodesFree")]
+    ImagefsInodesFree,
+    #[serde(rename = "pid.available")]
+    PidAvailable,
+}
+
+impl TryFrom<&str> for EvictionHardKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        serde_plain::from_str::<EvitionSignal>(&input).context(error::InvalidPlainValue {
+            field: "Eviction Hard key",
+        })?;
+        Ok(EvictionHardKey {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(EvictionHardKey, "EvictionHardKey");
+
+#[cfg(test)]
+mod test_kubernetes_eviction_hard_key {
+    use super::EvictionHardKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_eviction_hard_key() {
+        for ok in &[
+            "memory.available",
+            "nodefs.available",
+            "nodefs.inodesFree",
+            "imagefs.available",
+            "imagefs.inodesFree",
+            "pid.available",
+        ] {
+            EvictionHardKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_eviction_hard_key() {
+        for err in &["", "storage.available", ".bad", "bad.", &"a".repeat(64)] {
+            EvictionHardKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesThresholdValue represents a string that contains a valid kubernetes threshold value.
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesThresholdValue {
+    inner: String,
+}
+
+// Regular expression of Kubernetes quantitiy. i.e. 128974848, 129e6, 129M, 123Mi
+lazy_static! {
+    pub(crate) static ref KUBERNETES_QUANTITY: Regex =
+        Regex::new(r"^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$").unwrap();
+}
```

Comment:
This accepts nonsense like `50KiGTPeG911`.

Reply:
I take this regex from kubernetes quantity resource, and kubernetes used this regex to validate input value for `eviction-hard`, `kube-reserved` and some settings else. So should I still keep it or change it to more meaningful regex. https://github.com/kubernetes/kubernetes/blob/fcee7a01050652e54d2819b1942533d96e40f455/staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go#L133-L137

Btw, I had a discussion with Erikson around this validation about if we should be a strict gatekeeper here or if we just need to do a simple validation for that like it has to be a `SingleLineString`.

Reply:
I'd prefer to do stricter validation, because kicking it out at the API level results in a more immediately actionable error message.

Giving kubelet bad input will either cause it not to start, or it will ignore the setting. That's harder to diagnose because the error or warning will be buried in the journal.